### ISSUE TITLE
NeuralNetwork Giving back prediction classes that are non-existent in the training data

### ISSUE THREAD

**Issue by Xyclade** [haifengl/smile#71]
I'm not completely sure what is going on yet, but I have a setup where I've trained a neural network model with cross_entropy on a set of 2k points (I'm still collecting more data but wanted to give this a try).

They are labeled as classes [1...4]. However when I then predict for all my other data (+/- 20k records) I managed to get out a prediction label of 52. Should this be possible, or is my idea that it should only return prediction values of [1...4] correct?

**Comment by haifengl** [comment]
What is your network settings?

**Comment by Xyclade** [comment]
The network is set up as [4,100] for the layers

In the end it ran on 125k points and only 1 came back with this weird value, so I'll investigate a bit further. Just wanted to make sure that it should not return classifications outside the scope of the training data

**Comment by haifengl** [comment]
Can you share the setup code for the network? You have only two layers? Thanks!

**Comment by haifengl** [comment]
Do you find the reason?
